### ISSUE TITLE
Lines generated by Contours symbology to not have a $geometry

### ISSUE THREAD

**Issue by kannes** [qgis/QGIS#37965]
**Describe the bug**
Both lines created by the new Contour symbology option for raster layers (Contour Symbol and Index Contour Symbol) do not allow use of their `$geometry` in a Geometry Generator Symbol Layer.

**How to Reproduce**

1. Add a raster layer
2. Set its Symbology to Contours
3. Enter one of its symbols' (Contour Symbol or Index Contour Symbol) settings
4. Set it to a Geometry Generator
5. The lines will vanish. They will not appear if you change the GC's Geometry Type. If you enter the expression editor, you will see that `$geometry` results in `NULL`.

**QGIS and OS versions**

QGIS version | 3.15.0-Master | QGIS code revision | 1878fc7bfd
-- | -- | -- | --
Compiled against Qt | 5.15.0 | Running against Qt | 5.15.0
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.32.3 | Running against SQLite | 3.32.3
PostgreSQL Client Version | 12.3 | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.5 | QScintilla2 Version | 2.11.5
Compiled against PROJ | 6.3.1 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | Arch Linux

**Comment by saberraz** [comment]
I'd say it's a feature request :)

**Comment by kannes** [comment]
I had thought about that when filing and I disagree: The new Contours feature is lacking support for the existing GC symbology.

**Comment by nyalldawson** [comment]
Flipping this back to a bug -- either $geometry should be hidden here, or (better) we just make it work :)

**Comment by nyalldawson** [comment]
This was fixed by https://github.com/qgis/QGIS/pull/45165

**Comment by kannes** [comment]
Geometry Generator generated anaglyph 3D rendering here I come (at least trying)!

Thank you :)

**Comment by dontfearme22** [comment]
Sorry to re-open a old discussion, but I have been trying to extend this same line of thinking into using the generated geometries created by the raster contour symbology, to sample raster values in another layer, and finding it seemingly impossible - in ways that work with points generated in other ways. 

**Describe the bug**

Lines generated by the contour symbology option for rasters, when converted to point markers, do not sample raster values properly when used with the `raster_value()` function. 

**How to Reproduce**

1. Add a raster layer
2. Set its Symbology to Contours
3. Enter one of its symbols' (Contour Symbol or Index Contour Symbol) settings

(now the fun part) 

5. Convert to a marker line 
6. In the expression builder, try to write a `raster_value `query using `$geometry` on another raster layer. The resulting values are not consistent with the location of the generated point geometries. 

**Some screenshots**

I have noticed some odd and interesting behavior working with these generated geometries. 

First of all, @geometry - the recommended method, **does not work at all** when dealing with generated points inside a raster contour symbol. Using `make_point(x($geometry),(y($geometry))` as the point in the `raster_value()` function, and then tied to the font symbology produces these readouts:

<img width="365" alt="Screen Shot 2024-02-11 at 4 26 11 PM" src="https://github.com/qgis/QGIS/assets/47437740/e3376898-a984-4487-a23c-51326a86d635">

```
--x($geometry)
to_int(
raster_value('Churfirsten_120m_3ddad413_ac50_4d6a_8aa2_c36ef24549b4',1,make_point(
x($geometry),y($geometry)
))
)
--geom_to_wkt(make_point(
--x($geometry),y($geometry)
--))
```

While I have succeeded in varying attributes by size only one way:

<img width="524" alt="Screen Shot 2024-02-11 at 4 27 02 PM" src="https://github.com/qgis/QGIS/assets/47437740/b3720727-0413-4cae-8a9b-32e62fbd0f86">

```
--x($geometry)

(
raster_value('Churfirsten_120m_3ddad413_ac50_4d6a_8aa2_c36ef24549b4',1,make_point(
x($geometry),y($geometry)
))-1500)/20
--geom_to_wkt(make_point(
--x($geometry),y($geometry)
--))
```

However, in both cases  - the values do not correspond with the values of the DEM being queried:

<img width="878" alt="Screen Shot 2024-02-11 at 4 28 22 PM" src="https://github.com/qgis/QGIS/assets/47437740/fd7555e6-ef01-465b-b972-535f0dfceec0">

All these layers are in the same projection. I wonder if there is some complex error involving the CRS but do not know how to get around it. At a minimum, I would expect the points generated inside a raster contour layer, to operate the same as any other generated geometry. Would love to hear any thoughts or comments.

Qt version | 5.15.2
Python version | 3.9.5
GDAL/OGR version | 3.3.2
PROJ version | 8.1.1
EPSG Registry database version | v10.028 (2021-07-07)
GEOS version | 3.9.1-CAPI-1.14.2
SQLite version | 3.35.2
PDAL version | 2.3.0
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.6
QScintilla2 version | 2.11.5
OS version | macOS 11.7
